docs/src/guides/languages/postgresql.md

```diff
@@ -9,4 +9,301 @@ tags:
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+To apply and fix some formatting issues you can run `+format` target which will picks up directory
+where your Earthly file lies in as a source dir for formatting and run `+FORMAT` UDC target.
+Under the hood `+FORMAT` UDC target runs `sqlfluff-image` docker image,
+which contains the same configuration and setup which is applied during the `+check`.
+
+<!-- markdownlint-disable max-one-sentence-per-line -->
+!!! Note
+    Specific configuration of `sqlfluff` which is applied during the check and formatting you can find at the
+    [example](https://github.com/input-output-hk/catalyst-ci/blob/master/earthly/postgresql/.sqlfluff).
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+### Build
+
+```Earthfile
+build:
+    FROM +builder
+
+    ARG tag="latest"
+    ARG registry
+
+    DO ./../../earthly/postgresql+BUILD --image_name=example-db --tag=$tag --registry=$registry
+```
+
+With the `*.sql` files validation out of the way, we can finally build our PostgreSQL docker image.
+Since we need migration and seed data files,
+we'll inherit from the `builder` target.
+The actual image build process is pretty straight-forward
+and fully defined under the `+BUILD` UDC target.
+The only thing it is needed to specify is a few  arguments:
+
+* `tag` - the tag of the image, default value `latest`.
+* `registry` - the registry of the image.
+* `image_name` - the name of the image (required).
+
+### Run
+
+To run already builded docker image it is possible with the following `docker-compose.yml`
+
+```yaml
+version: "3"
+
+services:
+  postgres:
+    image: postgres:16
+    restart: unless-stopped
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+      POSTGRES_DB: postgres
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}"]
+      interval: 2s
+      timeout: 5s
+      retries: 10
+    ports:
+      - 5433:5432
+
+  example:
+    image: example-db:latest
+    environment:
+      # Required environment variables for migrations
+      - DB_HOST=${DB_HOST:-localhost}
+      - DB_PORT=5432
+      - DB_NAME=ExampleDb
+      - DB_DESCRIPTION=Example DB
+      - DB_SUPERUSER=postgres
+      - DB_SUPERUSER_PASSWORD=postgres
+      - DB_USER=example-dev
+      - DB_USER_PASSWORD=example-pass
+
+      - INIT_AND_DROP_DB=${INIT_AND_DROP_DB:-true}
+      - WITH_MIGRATIONS=${WITH_MIGRATIONS:-true}
+      - WITH_SEED_DATA=${WITH_SEED_DATA:-true}
+    ports:
+      - 5432:5432
+```
+
+There are 4 possible options how to run:
+
+* If `DB_HOST` env var established to `localhost`,
+PostgreSQL server runs as a part of the `example` service,
+otherwise will relies on remote PostgreSQL server connection
+(as an example already defined `postgres` service).
+* `INIT_AND_DROP_DB` env var defines to run initial initialization of the db with the clean state or not
+(could be omitted).
```

Comment:
Use the word `optional` instead.  But also put a description of what happens when the option is NOT specified.  Does it default, does it cause the container to behave differently? 

Same comment applies to each option below.

Reply:
fixed